### PR TITLE
More unit tests

### PR DESCRIPTION
**Pull request by ws233**
1. no more readwrite absoluteDataPath property.
2. Test for version
3. Test for initWithLanguage:engineMode
4. Tests for configEngine
5. Tests for resetEngine

### CODE REVIEW

**Comment by kevincon** [comment]
Thanks for adding these tests! Do you want me to merge and close this now or leave it open? Do you want to add any more tests?

**Comment by ws233** [comment]
I'm in progress with some more tests.
But I think it's better to spread them to different small commits.
Otherwise it will be to big and diffucalt to understand.

**Comment by ws233** [comment]
That's enough so far.

File: TesseractOCR/G8Tesseract.mm [hunk 1]
```diff
@@ -112,7 +113,10 @@ - (id)initWithLanguage:(NSString *)language
 
             _absoluteDataPath = [cachesPath stringByAppendingPathComponent:_absoluteDataPath].copy;
 
-            [self moveTessdataToCachesDirectoryIfNecessary];
+            BOOL succes = [self moveTessdataToCachesDirectoryIfNecessary];
+            if (succes == NO) {
```
Comment (kevincon) [inline TesseractOCR/G8Tesseract.mm]: Can you fix the typo here? It should be "success". After that I'll merge, everything else looks good!

**Comment by ws233** [comment]
Done.
You should also note, that G8Tesseract initializer will now return nil, if it's called with cashes related datapath and there is niegther tessdata folder in the Cashes directory nor App bundle.
That means it first tries to copy tessdata from the App bundle to Caches folder and start from there, if there is no tessdata folder yet. In the case there is also no tessdata in the App bundle it cannot copy anything and cannot start successfully. Therefore returns nil.

**Comment by kevincon** [comment]
Sounds good, thanks again for these contributions! Merging and closing.

**Pull request merged**